beacontools/scanner.py

```diff
@@ -100,13 +105,55 @@ def run(self):
             if event == LE_META_EVENT and subevent == EVT_LE_ADVERTISING_REPORT:
                 # we have an BLE advertisement
                 self.process_packet(pkt)
-
-    def toggle_scan(self, enable):
-        """Enable and disable BLE scanning."""
-        if enable:
-            command = "\x01\x00"
+        self.socket.close()
+
+    def set_scan_parameters(self, scan_type=ScanType.ACTIVE, interval_ms=10, window_ms=10,
+                            mac_type=BluetoothAddressType.RANDOM, filter_type=ScanFilter.ALL):
+        """"sets the le scan parameters
+            type     - ScanType.(PASSIVE|ACTIVE)
+            interval - ms between scans (valid range 2.5ms - 10240ms)
+                  !note: when interval and window are equal, the scan runs continuos
+            window   - ms scan duration (valid range 2.5ms - 10240ms)
+            own_type - Bluetooth address type BluetoothAddressType.(PUBLIC|RANDOM)
+                       PUBLIC = use device Bluetooth MAC address
+                       RANDOM = generate and use a random MAC address
+            filter   - ScanFilter.(ALL|WHITELIST_ONLY) only ALL is supported, which will
+                       return all fetched bluetooth packets (WHITELIST_ONLY is not supported,
+                       because OCF_LE_ADD_DEVICE_TO_WHITE_LIST command is not implemented)"""
+        interval_fractions = interval_ms / MS_FRACTION_DIVIDER
+        if interval_fractions < 0x0004 or interval_fractions > 0x4000:
+            raise ValueError(
+                "Invalid interval given {}, must be in range of 2.5ms to 10240ms!".format(
+                    interval_fractions))
+        window_fractions = window_ms / MS_FRACTION_DIVIDER
+        if window_fractions < 0x0004 or window_fractions > 0x4000:
+            raise ValueError(
+                "Invalid window given {}, must be in range of 2.5ms to 10240ms!".format(
+                    window_fractions))
+
+        if interval_fractions.is_integer() and window_fractions.is_integer():
```

Comment:
Why this if and the exception? IMO this is a little bit too much, ```window_fractions, interval_fractions = int(window_fractions), int(interval_fractions)``` should be perfectly fine.

Reply:
I was quiet motivated to raise early, but now I have changed it into the simple cast :)